manual_run/variables.py

```diff
@@ -0,0 +1,36 @@
+from typing import Callable, Dict, List
+
+
+class Variables:
+    # The number of student must match the number of students in the input CSV file
+    num_students = 146
+    # team size is the number of teams
+    team_size = 4
+
+    # Each dictionary should define the name of the column and each possible options for each column containing a non-integer value
+    attribute_options: Dict[str, List[str]] = {
+        "Q8": [
+            "In-person before or after class",
+            "In-person nights or weekends",
+            "On zoom",
+        ],
+        "Q4": [
+            "I am looking for a classmate to tutor me in BIOC 202",
+            "I am open to being a peer tutor or having a classmate tutor me in BIOC 202. I am uncertain if I should sign up as a tutor or tutee",
+            "I am interested in being a peer tutor in BIOC 202",
+        ],
+        "Q5": ["1", "2 to 3", "3+"],
+    }
+
+    attribute_handlers: Dict[str, Callable] = {
+        "z-score": lambda x: 1 if float(x) >= 0 else 0
+    }
+
+    # Enter the name of the columns from the provided CSV file
```

Comment:
Like all the columns in the provided CSV file? If it's all of them then why do we need this?

Reply:
These were simply the important data fields to return that relate to the group formation. In reality the doc provided had the following columns (most of which dont need to be returned to the prof to identify the newly formed groups): ResponseId, Q8, Lecture, Midterm 1, Assignment 1, Assignment 2, Assignment 3, Assignment 4, Assignment 5, Assignment 6, Current points, Final points, etc...

Reply:
So the goal is like "only deal with this column if it's named in this array"? If so, where is the logic that "filters" out the irrelevant columns?

Reply:
Given the provided attributes in the two dictionaries **attribute_options** and **attribute_handlers** then when a row is processed it will only process the data in those columns. Given that that information is already provided I have now redone this to automatically create those rows based on the data in the two dictionaries. So now the data_fields list is not present in the variables.py script. Here's how it looks now in the **create_csv** function inside of the manual_priority_run.py script:

```
data_fields = [["ResponseId"]]
        for key in Variables.attribute_options:
            data_fields[0].append(key)
        for key in Variables.attribute_handlers:
            data_fields[0].append(key)
        data_fields[0].append("TeamSizeViolation")
        data_fields[0].append("TeamId")
```